tests/test_reactive_plugin.py

```diff
@@ -53,101 +53,114 @@ def broken_charm_exe(tmp_path):
-            "reactive-charm-build-arguments": [
-                "--charm-argument",
-                "--charm-argument-with argument",
-            ],
-        }
-        self.plugin_properties = reactive_plugin.ReactivePluginProperties.unmarshal(spec)
-        part_spec = plugins.extract_part_properties(spec, plugin_name="reactive")
-        part = craft_parts.Part(
-            "foo", part_spec, project_dirs=project_dirs, plugin_properties=self.plugin_properties
-        )
-        project_info = craft_parts.ProjectInfo(
-            application_name="test",
-            project_dirs=project_dirs,
-            cache_dir=tmp_path,
-            project_name="fake-project",
-        )
-        part_info = craft_parts.PartInfo(project_info=project_info, part=part)
-
-        self._plugin = plugins.get_plugin(
-            part=part,
-            part_info=part_info,
-            properties=self.plugin_properties,
-        )
-
-    def test_get_build_package(self):
-        assert self._plugin.get_build_packages() == set()
-
-    def test_get_build_snaps(self):
-        assert self._plugin.get_build_snaps() == set()
-
-    def test_get_build_environment(self):
-        assert self._plugin.get_build_environment() == {}
-
-    def test_get_build_commands(self, tmp_path):
-        assert self._plugin.get_build_commands() == [
-            f"{sys.executable} -I {reactive_plugin.__file__} fake-project "
-            f"{tmp_path}/parts/foo/build {tmp_path}/parts/foo/install "
-            "--charm-argument --charm-argument-with argument"
-        ]
-
-    def test_invalid_properties(self):
-        with pytest.raises(pydantic.ValidationError) as raised:
-            reactive_plugin.ReactivePlugin.properties_class.unmarshal(
-                {"source": ".", "reactive-invalid": True}
-            )
-        err = raised.value.errors()
-        assert len(err) == 1
-        assert err[0]["loc"] == ("reactive-invalid",)
-        assert err[0]["type"] == "value_error.extra"
-
-    def test_validate_environment(self, charm_exe):
-        validator = self._plugin.validator_class(
-            part_name="my-part",
-            env=f"PATH={str(charm_exe.parent)}",
-            properties=self.plugin_properties,
-        )
+@pytest.fixture()
+def plugin_properties(tmp_path):
+    spec = {
+        "plugin": "reactive",
+        "source": str(tmp_path),
+        "reactive-charm-build-arguments": [
+            "--charm-argument",
+            "--charm-argument-with argument",
+        ],
+    }
+    plugin_properties = reactive_plugin.ReactivePluginProperties.unmarshal(spec)
+    return plugin_properties
+
+
+@pytest.fixture()
+def plugin(tmp_path, plugin_properties):
+    project_dirs = craft_parts.ProjectDirs(work_dir=tmp_path)
+    spec = {
+        "plugin": "reactive",
+        "source": str(tmp_path),
+        "reactive-charm-build-arguments": [
+            "--charm-argument",
+            "--charm-argument-with argument",
+        ],
+    }
```

Comment:
I don't like duplicating the `spec` here, but couldn't find other way of doing this.

Reply:
is just because of `tmp_path`? You could have a `spec` fixture that receives `tmp_path` and returns this dict

Reply:
good idea, did that!